tests/v2/actions/provisioning/creates.go

```diff
@@ -1071,3 +1071,28 @@ func DisableUpdateConfig(client *rancher.Client) {
 		return nil
 	})
 }
+
+// CreateProvisioningRKE1ClusterWithClusterTemplate provisions an rke1 cluster by using the rke1 template and revision ID and other values from the template.
+func CreateProvisioningRKE1ClusterWithClusterTemplate(client *rancher.Client, provider RKE1Provider, rke1Cluster *management.Cluster, clusterConfig *clusters.ClusterConfig, nodeTemplate *nodetemplates.NodeTemplate) (*management.Cluster, error) {
+	clusterName := namegen.AppendRandomString(provider.Name.String())
+	clusterResp, err := shepherdclusters.CreateRKE1Cluster(client, rke1Cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	if client.Flags.GetValue(environmentflag.UpdateClusterName) {
+		pipeline.UpdateConfigClusterName(clusterName)
+	}
+
+	var nodeRoles []nodepools.NodeRoles
+	for _, nodes := range clusterConfig.NodePools {
```

Comment:
if we only need nodePools, can we just pull those out from the clusterConfig instead of sending the whole object?

Reply:
clusterConfig also requires cluster template ID and revision ID. thats why its needed here. Adding additional functions just for nodePools can cause difficulty understanding the flow.

Reply:
but clusterConfig isn't used again in this function. So what I'm saying is, why require the entire object in the parameters if you only need nodePools from it for this function?

Reply:
sure, that was taking in the inspiration from https://github.com/rancher/rancher/blob/main/tests/v2/actions/provisioning/creates.go#L408 but I can just send in the noderoles instead to the function definition here.